myriad-scheduler/src/main/java/org/apache/myriad/Main.java

```diff
@@ -156,13 +157,19 @@ private void initProfiles(Injector injector) {
     ServiceProfileManager profileManager = injector.getInstance(ServiceProfileManager.class);
     Map<String, Map<String, String>> profiles = injector.getInstance(MyriadConfiguration.class).getProfiles();
     TaskUtils taskUtils = injector.getInstance(TaskUtils.class);
+    OfferUtils.setVcoreRatio(taskUtils.getNodeManagerVcoreRatio());
     if (MapUtils.isNotEmpty(profiles)) {
       for (Map.Entry<String, Map<String, String>> profile : profiles.entrySet()) {
         Map<String, String> profileResourceMap = profile.getValue();
-        if (MapUtils.isNotEmpty(profiles) && profileResourceMap.containsKey("cpu") && profileResourceMap.containsKey("mem")) {
-          Long cpu = Long.parseLong(profileResourceMap.get("cpu"));
+        if (MapUtils.isNotEmpty(profiles) &&
+            (profileResourceMap.containsKey("vcore") || profileResourceMap.containsKey("cpu")) &&
+            profileResourceMap.containsKey("mem")) {
+          Long vcore = profileResourceMap.containsKey("vcore") ?
+              Long.parseLong(profileResourceMap.get("vcore")) :
+              Long.parseLong(profileResourceMap.get("cpu"));
```

Comment:
Not sure why do we need to use vcore or cpu interchangeably here. If it is cpu and vcore ratio is defined then vcores = cpu /ratio and if vcores are defined then no need for ratio I think.
Should it be two separate params?